src/Raven.Server/Documents/Sharding/ShardedRequestHandler.cs

```diff
@@ -79,7 +82,26 @@ public override async Task WaitForIndexToBeApplied(TransactionOperationContext c
                     var db = await task;
                     tasks.Add(db.RachisLogIndexNotifications.WaitForIndexNotification(index, ServerStore.Engine.OperationTimeout));
                 }
-                await tasks.WhenAll();
+                await Task.WhenAll(tasks);
+            }
+        }
+
+        protected async Task WaitForExecutionOfRaftCommands(JsonOperationContext context, List<long> raftIndexIds)
+        {
+            using (var cts = CancellationTokenSource.CreateLinkedTokenSource(ServerStore.ServerShutdown))
+            {
+                var timeToWait = ServerStore.Configuration.Cluster.OperationTimeout.GetValue(TimeUnit.Milliseconds) * raftIndexIds.Count;
```

Comment:
cc @karmeli87 is this what we want?

Reply:
I think it is might be too much... let's have a max of 1 Minute? or a new configuration option?